### ISSUE TITLE
ENH: add `regex=False` option to pandas.Series.str.match and fullmatch (like in str.contains)

### ISSUE THREAD

**Issue by robna** [pandas-dev/pandas#48086]
### Feature Type

- [ ] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

pandas.Series.str.contains accepts the parameter `regex` which is a boolean defaulting to True, meaning that the provided pattern is treated as a regex expression. It is useful that this behaviour can be turned of, for instance if one tries to match a string which contains parentheses (which would be interpreted as regex groups)

Example:

`pd.Series.str.match('AB(CD EF)')` would be interpreted as regex with groups

...same for `pd.Series.str.fullmatch('AB(CD EF)')`. This is not always wanted. I.e. I would like to do a fullmatch on the literal string `'AB(CD EF)'`

It would work with: `pd.Series.str.contains('AB(CD EF)', regex=False)`, but then of course the matching is less strict.

### Feature Description

I don't know if its possible this way but perhaps copy how pd.Series.str.contains works with `regex=False` in the [pandas/core/strings/accessor.py](https://github.com/pandas-dev/pandas/commit/638a42fcc2acd66f75ac589a031dd533d5cbdd53#diff-b561abce692e1f6b2fb34ee594893c19ebb1694949d4398cc25f535fb133c943)

### Alternative Solutions

An alternative might by to escape relevant characters to avoid interpretation as regex control symbols. But this becomes unpractical if the match string is not directly entered as a string but obtained from a variable which contents are not fixed. Or when its used within a loop.

**Comment by tomaarsen** [comment]
Regarding your alternative solutions, you can rely on [`re.escape(...)`](https://docs.python.org/3/library/re.html#re.escape) to programmatically escape a string in preparation for an exact regex match.

**Comment by goeb86** [comment]
I agree with @tomaarsen , unless I am missing something. I would just use the `pd.Series.str.find` , or just use `re.escape`.

I'm not sure how strict the match needs to be, but either approach should work fine.

**Comment by robna** [comment]
The idea was that people might not know that `pd.Series.str.match` and  `pd.Series.str.fullmatch` are intended to be used with regex primarily, and may assume that they can apply it to non-regex strings in the same way they can do with  `pd.Series.str.contains`.
But for me the way with using `re.escape` is simple enough.
I will close this issue, maybe it will help some others too.

**Comment by MMK-IBSEN** [comment]
I ran into the same issue as @robna , took a while to troubleshoot but in the end I found out that the parentheses in my match string were the issue. If the regex=false was implemented in this function, it would have saved me a lot of time and worries.

The suggestion of @tomaarsen is of course a simpler way of achieving the same thing and I will be using that method from now on. Lesson learned the hard way I guess.
